Makefile

```diff
@@ -69,16 +71,22 @@ JSDOC=jsdoc
 LINT=.make-lint
 TEST=.make-test.js
 
+.PHONY: pre-generate-files-for-build
+pre-generate-files-for-build:
+	mkdir -p $(BUILD)
+	# create the version variable and use the current date + time for versioning
+	echo 'var version=$(shell date +%Y%m%d%H%M%S);' > $(BUILD)/makefile_variables.js
```

Comment:
Since this variable will be available in all the ermrestjs files, just to make sure it's not going to collide with any other variables, let's use a more specific name for it instead of just `version`. You could call it `buildTime` (`ermrestjsBuilTime`?) or `buildVersion` (`ermrestjsBuildVersion`?) or something more specific even.